k8s/upgrades/0.8.2-0.9.0/cstor/cstor-pool-update-082-090.yaml

```diff
@@ -0,0 +1,664 @@
+
+    {{- $message := printf "Successfully got details of CStorPool deployment {%s}." .UpgradeItem.name -}}
+    {{- $status :="" }}
+    {{- if eq .TaskResult.poolDeployment.labels "0.8.2" }}
+    {{- $status =.Config.successStatus.value }}
+    {{- else }}
+    {{- $status =.Config.failStatus.value }}
+    {{- end }}
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-deployment" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "CStorPool deployment version is not 0.8.2" true | saveAs "poolDeployment.verifyErr" .TaskResult | noop -}} 
+    {{- end }}
+---
+# upgrade-cstor-pool-082-090-get-cstorpool gets details of StoragePool CR.
+# This runtask is for get object info which will use in some runtask.
+# This should always run.
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-get-storagepool
+  namespace: default
+spec:
+  meta: |
+    id: StoragePool
+    apiVersion: openebs.io/v1alpha1
+    kind: StoragePool
+    action: get
+    objectName: {{ .UpgradeItem.name }}
+    disable: false
+  post: |
+    {{- jsonpath .JsonResult "{.metadata.labels.openebs\\.io/version}" | trim | saveAs "StoragePool.labels" .TaskResult | noop -}}
+
+    {{- $message := printf "Successfully got details of StoragePool {%s}." .UpgradeItem.name -}}
+    {{- $status :="" }}
+    {{- if eq .TaskResult.StoragePool.labels "0.8.2" }}
+    {{- $status =.Config.successStatus.value }}
+    {{- else }}
+    {{- $status =.Config.failStatus.value }}
+    {{- end }}
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-storagepool" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "StoragePool version is not 0.8.2" true | saveAs "StoragePool.verifyErr" .TaskResult | noop -}} 
```

Comment:
why do we throw verify error if subsequent tasks can disable based on their own version

Reply:
If we throw verify error how do we handle retrying an upgrade that had failed earlier